test/src/unittests/rhythm/test_rhythmextractor2013.py

```diff
@@ -0,0 +1,192 @@
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+
+
+from essentia_test import *
+from essentia.standard import MonoLoader, RhythmExtractor2013
+from math import fabs
+
+class TestRhythmExtractor2013(TestCase):
+
+    def testRegression(self):
+        audio = MonoLoader(filename=join(testdata.audio_dir, 'recorded', 'techno_loop.wav'))()
+        rhythm = RhythmExtractor2013()
+
+        bpm, _, _, _ ,_= rhythm(audio)
+        self.assertAlmostEqualFixedPrecision(bpm, 124.29265, 0) # exact value= 125.726791382
+
+    def runInstance(self, input, method="degara"):
+        print('TestRhythmExtractor2013: Warning - these tests are evaluated with high tolerances for error, please review these tests')
+
+        gen = VectorInput(input)
+        
+        # TODO differentiate Degara and Multinstance scenarios
+        rhythm = RhythmExtractor2013(method=method)
+        
+        #output all the outputs in the natural order in the std mode
+        return(rhythm(input))
+
+    def pulseTrain(self, bpm, sr, offset, dur):
+        from math import floor
+        period = int(floor(sr/(bpm/60.)))
+        size = int(floor(sr*dur))
+        phase = int(floor(offset*sr))
+
+        if phase > period:
+            phase = 0
+
+        impulse = [0.0]*size
+        for i in range(size):
+            if i%period == phase:
+                impulse[i] = 1.0
+        return impulse
+
+    def assertVectorWithinVector(self, found, expected, precision=1e-7):
+        for i in range(len(found)):
+            for j in range(1,len(expected)):
+                if found[i] <= expected[j] and found[i] >= expected[j-1]:
+                    if fabs(found[i] - expected[j-1]) < fabs(expected[j] - found[i]):
+                        self.assertAlmostEqual(found[i], expected[j-1], precision)
+                    else:
+                        self.assertAlmostEqual(found[i], expected[j], precision)
+
+    def _assertEqualResults(self, result, expected):
+        self.assertEqual(result[0], expected[0]) #bpm
+        self.assertEqualVector(result[1], expected[1]) # ticks
+        self.assertEqual(result[2], expected[2]) # confidence
+        self.assertEqualVector(result[3], expected[3]) # estimates
+        self.assertEqualVector(result[4], expected[4]) # bpmIntervals
+
+    #TODO The exact expected values for the result oof passing empty parameters is TBD
+    # (see testZero)
+    def testEmpty(self):
+        input = []
+        #expected = [0.0, [], 0.0, [], [],[]]# extra frame for confidence
+        #result = self.runInstance(input, method="multifeature")
+        #self.assertEqualVector(result, expected)
+        #result = self.runInstance(input, method="degara")
+        #self.assertEqualVector(result, expected)
+        #print(result) 
+
+    def testZero(self):
+        input = array([0.0]*10*1024) # 100 frames of size 1024
+        expected = [0.0, [], 0.0, [], [],[]] # extra frame for confidence
```

Comment:
that comment was intended to differentiate it from the interface of Rhythm extractor which has 4 outputs,
The extra output for rhythmextractor2013 is the confidence factor, but I guess this comment doens add value since the algorithm is in a separate file.